cpp/src/replace/replace.cu

```diff
@@ -134,3 +135,96 @@ gdf_error gdf_find_and_replace_all(gdf_column*       col,
 {
   return find_and_replace_all(col, old_values, new_values);
 }
+
+namespace{ //anonymous
+
+template <typename Type>
+__global__
+void replace_nulls_with_scalar(gdf_size_type size, Type* out_data, gdf_valid_type * out_valid, const Type *in_data_scalar) 
+{
+  int tid = threadIdx.x;
+  int blkid = blockIdx.x;
+  int blksz = blockDim.x;
+  int gridsz = gridDim.x;
+
+  int start = tid + blkid * blksz;
+  int step = blksz * gridsz;
+
+  for (int i=start; i<size; i+=step) {
+    out_data[i] = gdf_is_valid(out_valid, i)? out_data[i] : *in_data_scalar;
+  }
+}
+
+
+template <typename Type>
+__global__
+void replace_nulls_with_column(gdf_size_type size, Type* out_data, gdf_valid_type* out_valid, const Type *in_data) 
+{
+  int tid = threadIdx.x;
+  int blkid = blockIdx.x;
+  int blksz = blockDim.x;
+  int gridsz = gridDim.x;
+
+  int start = tid + blkid * blksz;
+  int step = blksz * gridsz;
+
+  for (int i=start; i<size; i+=step) {
+    out_data[i] = gdf_is_valid(out_valid, i)? out_data[i] : in_data[i];
+  }
+}
+
+
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief Functor called by the `type_dispatcher` in order to invoke and instantiate
+ *        `replace_nulls` with the apropiate data types.
+ */
+/* ----------------------------------------------------------------------------*/
+struct replace_nulls_kernel_forwarder {
+  template <typename col_type>
+  void operator()(gdf_size_type           nrows,
+                  gdf_size_type           new_values_length,
+                  void*            d_col_data,
+                  gdf_valid_type*  d_col_valid,
+                  const void*      d_new_value)
+  {
+    const size_t grid_size = nrows / BLOCK_SIZE + (nrows % BLOCK_SIZE != 0);
+    if (new_values_length == 1) {
+      replace_nulls_with_scalar<<<grid_size, BLOCK_SIZE>>>(nrows,
+                                            static_cast<col_type*>(d_col_data),
+                                            (d_col_valid),
+                                            static_cast<const col_type*>(d_new_value)
+                                            );
+    } else if(new_values_length == nrows) {
+      replace_nulls_with_column<<<grid_size, BLOCK_SIZE>>>(nrows,
+                                            static_cast<col_type*>(d_col_data),
+                                            (d_col_valid),
+                                            static_cast<const col_type*>(d_new_value)
+                                            );
+      
+    }
+  }
+};
+
+} //end anonymous namespace
+
+gdf_error gdf_replace_nulls(gdf_column* col_out, const gdf_column* new_values_column)
```

Comment:
`new_values_column` name doesn't agree with name in `functions.h`.

Reply:
done